src/c++/perf_analyzer/docs/examples/profile.py

```diff
@@ -27,35 +27,78 @@
 import argparse
 import json
 import subprocess
+from dataclasses import dataclass
 from itertools import pairwise
 from pathlib import Path
+from typing import Optional
 
 import numpy as np
 
-TEMP_INPUT_FILE = "temp_input_data.json"
+INPUT_FILENAME = "generated_input_data.json"
 
+TITLE = "\n[ BENCHMARK SUMMARY ]\n"
+PROMPT_SIZE = "  Prompt size: {}"
+FIRST_TOKEN_LATENCY = "Average first-token latency: {:.4f} sec"
+T2T_LATENCY = "Average total token-to-token latency: {:.4f} sec"
 
-def load_profile_data():
-    with open("profile_export.json") as f:
+
+@dataclass
+class ProfileResults:
+    prompt_size: int
+    avg_first_token_latency: int
+    avg_total_t2t_latency: int
+    avg_periodic_t2t_latencies: Optional[list[int]] = None
+
+
+def load_json_data(filename):
+    with open(filename) as f:
         return json.load(f)
 
 
-def print_benchmark_summary(results):
-    output = "\n[ Benchmark Summary ]"
-    for prompt_size, avg_first_token_latency, avg_token_to_token_latency in results:
-        output += (
-            f"\n  Prompt size: {prompt_size}, "
-            f"Average first-token latency: {avg_first_token_latency:.4f} sec"
-        )
-        output += (
-            f", Average token-to-token latency: {avg_token_to_token_latency:.4f} sec"
-            if avg_token_to_token_latency
-            else ""
-        )
-    print(output)
+def save_json_data(data, filename):
+    with open(filename, "w") as f:
+        json.dump(data, f)
+
+
+def get_postfix(args, prompt_size):
+    """Generate postfix for profile export filename and plot.
+
+    e.g.
+      - trtllm-prompt100-maxtokens256
+      - trtllm-prompt100-periodic1_100_1-period32-maxtokens1024
+    """
+    postfix = f"{args.model}-prompt{prompt_size}-"
+    if args.periodic_concurrency_range:
+        start, end, step = args.periodic_concurrency_range
+        postfix += f"periodic{start}_{end}_{step}-period{args.request_period}-"
+    postfix += f"maxtokens{args.max_tokens}"
+    return postfix
+
+
+def get_export_filename(args, prompt_size):
+    postfix = get_postfix(args, prompt_size)
+    filename = f"profile_export-{postfix}.json"
+    return filename
+
+
+def get_plot_filename(args, prompt_size):
+    postfix = get_postfix(args, prompt_size)
+    filename = f"inflight_batching_benchmark-{postfix}.png"
+    return filename
 
 
-def plot_results(latencies):
+def print_benchmark_summary(profile_results):
+    output = [TITLE]
+    for pr in profile_results:
+        line = [PROMPT_SIZE.format(pr.prompt_size)]
```

Comment:
Why are we making format calls here? Can we make these fstrings?

Reply:
The main reason is that variables were too long and it looked messy with all the strings and variables embedded together.

Reply:
the modest speed improvement is not worth the readability loss. 
thanks for the info.